### PR TITLE
27 create admin log insign up view

### LINKED ISSUE #27: Create Admin log in/sign up view
# REQUIREMENTS
- Add sign in and sign up view
- Use device to make the log in and sign up
# DESIGN
image

### PR DESCRIPTION
**Pull request by femm34**
# Description

What did you implemented:
 - We use the gem 'devise' to be able to do a registration and login
 - Views of the 'Sign in', 'Sign up' and 'Forgot your password?'

 Why did you implemented this:
 - We use devise to facilitate user authentication
 - To improve the user experience

  ## Screenshots
image
image
image
image

### CODE REVIEW

File: app/controllers/admins/confirmations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  # The path used after resending confirmation instructions.
+  # def after_resending_confirmation_instructions_path_for(resource_name)
+  #   super(resource_name)
+  # end
+
+  # The path used after confirmation.
+  # def after_confirmation_path_for(resource_name, resource)
+  #   super(resource_name, resource)
+  # end
```
Comment (YilKanda) [inline app/controllers/admins/confirmations_controller.rb]: If you don't use this code, you can delete it.

File: app/controllers/admins/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  # def failure
+  #   super
+  # end
+
+  # protected
+
+  # The path used when OmniAuth fails
+  # def after_omniauth_failure_path_for(scope)
+  #   super(scope)
+  # end
```
Comment (YilKanda) [inline app/controllers/admins/omniauth_callbacks_controller.rb]: If you don't use this code, you can delete it.

File: app/controllers/admins/passwords_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  # protected
+
+  # def after_resetting_password_path_for(resource)
+  #   super(resource)
+  # end
+
+  # The path used after sending reset password instructions
+  # def after_sending_reset_password_instructions_path_for(resource_name)
+  #   super(resource_name)
+  # end
```
Comment (YilKanda) [inline app/controllers/admins/passwords_controller.rb]: If you don't use this code, you can delete it.

File: app/controllers/admins/registrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  # GET /resource/cancel
+  # Forces the session data which is usually expired after sign
+  # in to be expired now. This is useful if the user wants to
+  # cancel oauth signing in/up in the middle of the process,
+  # removing all OAuth session data.
+  # def cancel
+  #   super
+  # end
+
+  # protected
```
Comment (YilKanda) [inline app/controllers/admins/registrations_controller.rb]: If you don't use this code, you can delete it.

File: app/controllers/admins/registrations_controller.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+  # The path used after sign up.
+  # def after_sign_up_path_for(resource)
+  #   super(resource)
+  # end
+
+  # The path used after sign up for inactive accounts.
+  # def after_inactive_sign_up_path_for(resource)
+  #   super(resource)
+  # end
```
Comment (YilKanda) [inline app/controllers/admins/registrations_controller.rb]: If you don't use this code, you can delete it.

File: app/controllers/admins/sessions_controller.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  # def destroy
+  #   super
+  # end
+
+  # protected
+
+  # If you have extra params to permit, append them to the sanitizer.
+  # def configure_sign_in_params
+  #   devise_parameter_sanitizer.permit(:sign_in, keys: [:attribute])
+  # end
```
Comment (YilKanda) [inline app/controllers/admins/sessions_controller.rb]: If you don't use this code, you can delete it.

File: app/controllers/admins/unlocks_controller.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+  # The path used after sending unlock password instructions
+  # def after_sending_unlock_instructions_path_for(resource)
+  #   super(resource)
+  # end
+
+  # The path used after unlocking the resource
+  # def after_unlock_path_for(resource)
+  #   super(resource)
+  # end
```
Comment (YilKanda) [inline app/controllers/admins/unlocks_controller.rb]: If you don't use this code, you can delete it.

File: db/migrate/20220325221841_devise_create_admins.rb [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+
```
Comment (YilKanda) [inline db/migrate/20220325221841_devise_create_admins.rb:38]: If you don't use this code, you can delete it.

File: app/controllers/admins/confirmations_controller.rb (hunk 1 above)
Comment (femm34) [reply app/controllers/admins/confirmations_controller.rb]: @YilKanda thank u, i will make these changes

File: app/controllers/admins/omniauth_callbacks_controller.rb (hunk 2 above)
Comment (femm34) [reply app/controllers/admins/omniauth_callbacks_controller.rb]: @YilKanda thank u, i will make these changes :smile:

File: app/controllers/admins/passwords_controller.rb (hunk 3 above)
Comment (femm34) [reply app/controllers/admins/passwords_controller.rb]: @YilKanda thanks i will make these changes :smile_cat:

File: db/migrate/20220325221841_devise_create_admins.rb [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
```
Comment (YilKanda) [inline db/migrate/20220325221841_devise_create_admins.rb:36]: Code that you do not use, delete it.

File: db/migrate/20220325221841_devise_create_admins.rb [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+
+      t.timestamps null: false
+    end
+
+    add_index :admins, :email,                unique: true
+    add_index :admins, :reset_password_token, unique: true
+    # add_index :admins, :confirmation_token,   unique: true
+    # add_index :admins, :unlock_token,         unique: true
```
Comment (YilKanda) [inline db/migrate/20220325221841_devise_create_admins.rb:45]: Code that you do not use, delete it.

**Review by YilKanda — APPROVED** [review]

**Review by brocha05 — APPROVED** [review]
LGTM

**Review by Frcomparan — APPROVED** [review]
LGTM

**Pull request merged**